### PR TITLE
Mast cloudaccess docs update

### LINKED ISSUE #2234: update readthedocs to reflect new anonymous cloud access for MAST data
Adrian Lucy noted that the current documentation still describes cloud access using a cloud account, but this should be updated now that anonymous cloud access is available to access our MAST databases:

https://astroquery.readthedocs.io/en/latest/mast/mast.html#cloud-data-access

I can tackle this this week.

### PR DESCRIPTION
**Pull request by jaymedina**
Closes #2234

### CODE REVIEW

File: docs/mast/mast.rst [hunk 1]
```diff
@@ -343,26 +343,21 @@ the *local_path* keyword argument.
...
 in `public S3 buckets <https://registry.opendata.aws/collab/stsci/>`__.
 
-Using AWS resources to process public data requires an `AWS account <https://aws.amazon.com/>`__ and associated
-`credentials file <https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html>`__. The `boto3
-<https://boto3.amazonaws.com/v1/documentation/api/latest/index.html>`__ library is also required as it handles
-connections to the AWS servers. Instructions for creating AWS credentials are available `here
-<https://stackoverflow.com/questions/21440709/how-do-i-get-aws-access-key-id-for-amazon>`__. Data transfer charges
-are the responsibility of the requester  (see `request pricing <https://aws.amazon.com/s3/pricing/>`__), however
-transfers are free within the US-East AWS region.
+Using AWS resources to process public data no longer requires an `AWS account for all AWS regions. To enable cloud data access for the Hubble, Kepler, and TESS missions, follow the steps below:
```
Comment (bsipocz) [inline docs/mast/mast.rst:349]:
```suggestion
Using AWS resources to process public data no longer requires an AWS account for all AWS regions. To enable cloud data access for the Hubble, Kepler, and TESS missions, follow the steps below:
```

File: docs/mast/mast.rst [hunk 2]
```diff
@@ -343,26 +343,21 @@ the *local_path* keyword argument.
...
 
-Cload data access is enabled using the `~astroquery.mast.ObservationsClass.enable_cloud_dataset` function, which
-will cause AWS to become the prefered source for data access until it is disabled
+Enable cloud data access via the `~astroquery.mast.ObservationsClass.enable_cloud_dataset` function, which
+will cause AWS to become the preferred source for data access until it is disabled
```
Comment (bsipocz) [inline docs/mast/mast.rst:352]: maybe rephrase?
```suggestion
You can enable cloud data access via the `~astroquery.mast.ObservationsClass.enable_cloud_dataset` function, which sets AWS to become the preferred source for data access as opposed to on-premise MAST until it is disabled with 
```

File: docs/mast/mast.rst [hunk 3]
```diff
@@ -343,26 +343,21 @@ the *local_path* keyword argument.
...
 
 To directly access a list of cloud URIs for a given dataset, use the `~astroquery.mast.ObservationsClass.get_cloud_uris`
-function, however when cloud access is enabled, the standatd download function
-`~astroquery.mast.ObservationsClass.download_products` will preferentially pull files from AWS when they are avilable.
-There is also a ``cloud_only`` flag, which when set to True will cause all data products not available in the
-cloud to be skipped.
+function (Python will prompt you to enable cloud access if you haven't already).
+
+When cloud access is enabled, the standard download function
+`~astroquery.mast.ObservationsClass.download_products` will preferentially pull files from AWS when they are available.
```
Comment (bsipocz) [inline docs/mast/mast.rst:359]: I think technical texts like this reads better when written in the present tense rather than future as we're not really describing a planned, future experiment or such but describing an already existing behaviour. But I can stand corrected.

File: docs/mast/mast.rst [hunk 4]
```diff
@@ -343,26 +343,21 @@ the *local_path* keyword argument.
...
+When set to `True`, the ``cloud_only`` parameter in `download_products` will cause all data products not available in the cloud to be skipped.
```
Comment (bsipocz) [inline docs/mast/mast.rst:360]: You can't use just the function name to be linked, it's either single backticks to the full namespace as in the previous line, or double backticks to highlight but don't link. Either is fine by me, but the current one is causing a docs build failure.

**Review by bsipocz — CHANGES_REQUESTED** [review]
Test failure is a red herring, but the doc build failure is related to this PR.

File: docs/mast/mast.rst (hunk 2 above)
Comment (jaymedina) [reply docs/mast/mast.rst:352]: Yes this sounds better

**Comment by bsipocz** [comment]
@jaymedina - what are you trying to do with the last commit? Rerunning the CI with dummy commits without fixing the issue it failed on won't work. The issue was:

`/home/docs/checkouts/readthedocs.org/user_builds/astroquery/checkouts/2235/docs/mast/mast.rst:357: WARNING: Inline interpreted text or phrase reference start-string without end-string.`

So I would suggest adding the closing backtick that was removed in the second to last commit, and squashing the last 3 together in a rebase. Then all should work and we can merge.

**Comment by jaymedina** [comment]
@bsipocz forgot to close one of my strings. Looks like the build is passing now. It turns out the issue wasn't with the second to last commit, it was a string I forgot to close a couple lines below that. Let me know if you need anything else, thanks for looking over this.

**Comment by bsipocz** [comment]
Yep, that one wasn't closed in the commit I mentioned. 

If you could squash the last three ones it would be great, otherwise it's all looking good.

**Comment by jaymedina** [comment]
I tried rebasing and squashing using the documentation here (https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#how-to-squash) and got no errors but nothing seemed to happen in this PR, so I tried running this command which seemed to mess things up locally: 

```
git reset --soft HEAD~3 &&
git commit
```

Does the squashing need to be done for this PR or can it be merged as is? I'll familiarize myself with these git commands offline before the next PR.

**Comment by bsipocz** [comment]
I'll do the squash here then and merge it.


As for practice: My favourite resource both for practice and teach git branch management is https://learngitbranching.js.org/, it has a ton of scenarios (some are very much different than our workflow, but nevertheless, I think it gives a very solid understanding and muscle memory of the underlying logic).

For this case what I would do is (there are other ways, but with rebase you can basically fix any issues as it's not necessary to rebase on the latest `main` branch, etc.):

- `git rebase -i HEAD~5` ---> brings up editor where I would set the last two lines to `f` for fix-up. Save and exit. There won't be any conflicts.
- `git push jaymedina HEAD:mast-cloudaccess-docs-update --force` -- > force push back to this branch.

**Comment by bsipocz** [comment]
> and got no errors but nothing seemed to happen in this PR,

I guess you didn't force push back, it's necessary as there won't be any automated sync between branches, especially if their histories differ, and you have to force push explicitly.

**Review by bsipocz — APPROVED** [review]

**Comment by jaymedina** [comment]
I'll look into this page, thanks very much :) 

I did a `git push --force main [this-branch]` but for some reason nothing happened after that, so I'm not sure what happened. The only file in the `.git/git-rebase/` folder was called `git-merge-todo` or something like that, which I updated with a simple commit description but that didn't seem to do anything

**Comment by bsipocz** [comment]
You basically never need to manually do anything within the .git directory, I guess there might have been another reference, etc so your edit was not picked up. (Also, sometimes you may moved away from the branch locally, then if you push the branch it indeed have them identical (and thus I tend to always push HEAD explicitly into the remote branch)
